core/domain/prod_validation_jobs_one_off.py

```diff
@@ -6183,6 +6184,83 @@ def _get_external_instance_custom_validation_functions(cls):
         return [cls._validate_reference]
 
 
+class PlatformParameterModelValidator(BaseModelValidator):
+    """Class for validating PlatformParameterModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        return r'^[A-Za-z0-9_]+$'
```

Comment:
Why do we need the override though, wouldn't the regex defined in BaseModelValidator work? Here and below.

Reply:
We need to loosen the constraints on id length because the id (name) of the params are developer-specified (for example `A_Long_Feature_Name_Of_Frontend`), which can be much longer than the original length limit defined in BaseModelValidator.

Reply:
Can we have some max length constraint?

Reply:
I think you should have a max length constraint and get developers to be concise.

In general, have constraints for everything. We have run into problems with this before because sometimes we use IDs for one model to form part of other IDs.

Reply:
@ankita240796 it might be worth updating the one for ConfigProperty too, along similar lines?

Reply:
Having a max length seems like a good idea, and we can use a longer one here to accommodate long feature names. Anything longer than, say, 100 characters probably ought to be shortened, anyway.

Reply:
Done, the length limit is set to 100.

Reply:
Yes, makes sense. @MegrezZhu would it be possible to add the same for config property model validator as well - if not can you please create an issue to keep track?

Reply:
Done, added the same constraints on id of config properties (Sorry for the late reply, I somehow missed this thread...)